supernovacontroller/sequential/i3c.py

```diff
@@ -368,6 +368,33 @@ def target_update_address(self, current_address, new_address):
             result = (False, responses[0]["errors"])
 
         return result
+
+    def trigger_target_reset_pattern(self):
+        """
+        Triggers the target reset pattern on the I3C bus.
+
+        It indicates the targets to execute the reset action configured via the RSTACT CCC.
+
+        Returns:
+        tuple: A tuple containing two elements:
+            - The first element is a Boolean indicating the success (True) or failure (False) of the operation.
+            - The second element is either None indicating success, or an error message
+                detailing the failure, obtained from the device's response.
+        """
+        try:
+            responses = self.controller.sync_submit([
+                lambda id: self.driver.i3cTriggerTargetResetPattern(id)
+            ])
+        except Exception as e:
+            raise BackendError(original_exception=e) from e
+
+        response = responses[0]
+        if response["usb_result"] == "CMD_SUCCESSFUL" and response["manager_result"] == "I3C_CONTROLLER_MGR_NO_ERROR" and "NO_TRANSFER_ERROR" in response["driver_result"]:
+            result = (True, None)
+        else:
+            result = (False, response["descriptor"]["errors"])
```

Comment:
Are we sure of this?

When the target is connected everything is correct, but, I tried to run the tests without the device connected and one of the test raised an exception:
```
======================================================================
ERROR: test_trigger_target_reset_pattern (__main__.TestSupernovaController)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\jpons\Binho\BMC\SupernovaController\tests\test.py", line 608, in test_trigger_target_reset_pattern
    (success, result) = i3c.trigger_target_reset_pattern()
  File "c:\users\jpons\binho\bmc\supernovacontroller\supernovacontroller\sequential\i3c.py", line 396, in trigger_target_reset_pattern
    result = (False, response["descriptor"]["errors"])
KeyError: 'descriptor'

======================================================================
```

Inspecting the response, the SDK is giving this:
`{'id': 9, 'command': 25, 'name': 'I3C TRIGGER TARGET RESET PATTERN', 'usb_result': 'CMD_SUCCESSFUL', 'manager_result': 'I3C_CONTROLLER_MGR_NO_ERROR', 'driver_result': ['NACK_ERROR']}`

Reply:
Fixed and prettied the error handling :)